codemagic.yaml

```diff
@@ -0,0 +1,125 @@
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      cache:
+        cache_paths:
+          - $HOME/Library/Caches/CocoaPods
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'develop'
+            include: true
+            source: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match Ad-hoc
+          script: bundle exec fastlane syncAdHocStagingCodeSigning
+        - name: Build App and Distribute to Firebase
+          script: bundle exec fastlane buildStagingAndUploadToFirebase
+      artifacts:
+        - ./*.ipa
+        - ./*.dSYM.zip
+    deploy_production_firebase:
+      name: Deploy Production Build To Firebase
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      cache:
+        cache_paths:
+          - $HOME/Library/Caches/CocoaPods
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'release/*'
+            include: true
+            source: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match Ad-hoc
+          script: bundle exec fastlane syncAdHocProductionCodeSigning
+        - name: Build App and Distribute to Firebase
+          script: bundle exec fastlane buildProductionAndUploadToFirebase
+      artifacts:
+        - ./Output/*.ipa
+        - ./Output/*.dSYM.zip
+    deploy_app_store:
+      name: Deploy Build To App Store
+      instance_type: mac_mini_m1
+      environment:
+        groups:
+          - fastlane
+        xcode: latest
+        cocoapods: default
+      cache:
+        cache_paths:
+          - $HOME/Library/Caches/CocoaPods
+      triggering:
+        events:
+          - push
+        branch_patterns:
+          - pattern: 'main'
+            include: true
+            source: true
+      scripts:
+        - name: Install bundle
+          script: bundle install --path vendor/bundle
+        - name: Install Pods Dependencies
+          script: bundle exec pod install
+        - name: Build and Test
+          script: bundle exec fastlane buildAndTest
+        - name: Match AppStore
+          script: bundle exec fastlane syncAppStoreCodeSigning
+        - name: Build App and Distribute to AppStore
+          script: bundle exec fastlane buildAndUploadToAppStore
+      artifacts:
+        - ./Output/*.ipa
```

Comment:
Why don't we have the `dSYM.zip` file?

Reply:
Because we have no custom script for this artifact, I removed this

Reply:
![Screenshot 2023-02-09 at 15 30 06](https://user-images.githubusercontent.com/24598204/217758226-a29c4e05-2a6e-40ca-b10c-06a97743747a.png)
![Screenshot 2023-02-09 at 15 30 25](https://user-images.githubusercontent.com/24598204/217758244-bde74dd0-27f3-4d33-925d-f1357be46e9d.png)

Reply:
> we have no custom script for this artifact

What do you mean by this?

We have this #351, for codemagic, I think you can follow this: https://docs.codemagic.io/knowledge-others/upload-custom-artifacts/

Reply:
Yes, but the Fastlane have take care of uploading dSYM. As I understand we use artifact for case using Bitrise,CodeMagic to upload ipa or dsym itself

Reply:
@ducbm051291 I know that fastlane will upload builds and dsym to Firebase/TestFlight. But using artifacts is a different story. Currently, if you use only fastlane, can you get the file `ipa` and `dsym`? Sometimes, the CI can not upload dsym files to Firebase. How can you fix that issue?

Reply:
Ok, I got it now ;)

Reply:
Updated in 912a958

![Screenshot 2023-02-10 at 13 21 04](https://user-images.githubusercontent.com/24598204/218017968-26e8fab3-4bbb-4b73-8654-bc9ac6b0162a.png)